tests/waku_archive/test_retention_policy.nim

```diff
@@ -53,6 +54,51 @@ suite "Waku Archive - Retention policy":
 
     ## Cleanup
     (waitFor driver.close()).expect("driver to close")
+  
+  test "size retention policy - windowed message deletion":
+    ## Given
+    let
+      # in megabytes
+      sizeLimit:float = 0.05
+      excess = 123
+
+    let driver = newTestArchiveDriver()
+
+    let retentionPolicy: RetentionPolicy = SizeRetentionPolicy.init(size=sizeLimit)
+
+    ## When
+
+    var putFutures = newSeq[Future[ArchiveDriverResult[void]]]()
+
+    for i in 1..excess:
+        let msg = fakeWakuMessage(payload= @[byte i], contentTopic=DefaultContentTopic, ts=Timestamp(i))
+        putFutures.add(driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp))
+
+    # waitFor is used to synchronously wait for the futures to complete.
+    discard waitFor allFinished(putFutures)
+
+    ## Then
+    # calculate the current database size
+    var pageSize = (waitFor driver.getPagesSize()).tryGet()
+    var pageCount = (waitFor driver.getPagesCount()).tryGet()
+    var sizeDB = float(pageCount * pageSize) / (1024.0 * 1024.0)
+    # execute policy if the current db size oveflows 
+    if sizeDB >= sizeLimit:
+      require (waitFor retentionPolicy.execute(driver)).isOk()
```

Comment:
This is great indeed! However, would it be possible to enforce a `require` instead?
```suggestion
    require sizeDB >= sizeLimit
    require (waitFor retentionPolicy.execute(driver)).isOk()
```